### PR TITLE
chore: update lighthouse setup

### PR DESCRIPTION
**Pull request by anpa**
## Problem

We are using lighthouse to audit the theme for accessibility issues. Our current setup relies on a Lighthouse CI Action. Some issues with this setup:
- The current output is difficult to read;
- Running it locally is not straightforward;
- We have less control over the version of lighthouse;
- It's currently not possible to mute known issues/errors for specific pages;

## Description

This PR creates a node script that runs lighthouse as a [node module](https://github.com/GoogleChrome/lighthouse#using-the-node-module). The script acts as a wrapper to lighthouse so that:
- It customizes the output to provide more details (including the issue, url, selector and html element that failed the audit);
- Developers can test accessibility by running the script locally;
- Upgrading lighthouse is done by bumping its version in `package.json`;
- The github action uses the same node script;
- It allows muting/turning some errors into warnings for specific selectors on specific pages;

### To run the script

- **Dev mode** - It runs the accessibility audits on the preview of the local theme on a specific account. It requires `zat theme preview` to be running;

```
yarn test-a11y -d
```

- **CI mode** - It runs the accessibility audits on the live theme of a specific account.
```
yarn test-a11y
```

## Future improvements:
- Use `axe-core` instead of lighthouse;

## Screenshots

Screenshot 2022-12-23 at 14 11 57

## Checklist

- [x] :green_book: all commit messages follow the [conventional commits](https://conventionalcommits.org/) standard
- [ ] :nail_care: SASS files are compiled
- [ ] :arrow_left: changes are compatible with RTL direction
- [ ] :wheelchair: changes are accessible
- [ ] :memo: changes are tested in Chrome, Firefox, Safari and Edge
- [ ] :iphone: changes are responsive and tested in mobile
- [ ] :+1: PR is approved by @zendesk/vikings

### CODE REVIEW

**Review by sergioazevedo — APPROVED** [review]

**Review by luis-almeida — APPROVED** [review]
I think it looks really really good!! 💪🏼 
Just added a few minor comments so feel free to disregard what doesn't make sense.

Seems to be running consistently under `5mins` so that's quite good. If later on we add more checks and it starts taking considerably more time, we can always try to run it in child processes and parallelize each page check.

File: README.md [hunk 1]
```diff
@@ -98,16 +98,107 @@ Note: Zendesk App Tools [theme preview](#publishing-your-theme) currently does n
...
+2. Set `end_user_email`, `end_user_password`, `subdomain` and `urls` as environment variables and run the accessibility audits in CI mode i.e.:
+
+```console
+end_user_email=<EMAIL> \
+end_user_password=<PASSWORD> \
+subdomain=<SUBDOMAIN> \
+urls="
+    https://<SUBDOMAIN>.zendesk.com/hc/en-us/
+    https://<SUBDOMAIN>.zendesk.com/hc/en-us/requests/new
+    https://<SUBDOMAIN>.zendesk.com/hc/en-us/requests" \
```
[the remark this answers is not in the document]
Comment (anpa) [reply README.md:164]: Good point! I couldn't find a way to pass these secrets to the github action other than using env variables: https://github.com/zendesk/copenhagen_theme/blob/35aa36c7c8d71327b8d989839a4abed4e84a66d7/.github/workflows/lighthouse.yml#L38-L41

I'll give it a try in a follow up PR. I'd like to also make a clearer separation between admin and end-user credentials.

File: package.json [hunk 2]
```diff
@@ -15,9 +17,13 @@
     "@semantic-release/changelog": "6.0.2",
     "@semantic-release/exec": "6.0.3",
     "@semantic-release/git": "10.0.1",
+    "dotenv": "16.0.3",
     "husky": "8.0.2",
     "js-yaml": "4.1.0",
-    "semantic-release": "19.0.5"
+    "lighthouse": "9.6.8",
```
[the remark this answers is not in the document]
Comment (anpa) [reply package.json]: Thanks for pointing it out! It seems like an easy upgrade for this use case. I added it to this PR 🎉

**Comment by anpa** [comment]
Thanks for the review @luis-almeida! I've addressed your feedback in the last two commits. Do you mind having another look pls? 🙏

**Review by luis-almeida — APPROVED** [review]
👍🏼 !

**Pull request merged**